src/marqo/core/unstructured_vespa_index/unstructured_document.py

```diff
@@ -93,6 +95,14 @@ def from_marqo_document(cls, document: Dict, filter_string_max_length: int) -> "
             elif isinstance(value, float):
                 instance.fields.float_fields[key] = value
                 instance.fields.score_modifiers_fields[key] = value
+            elif isinstance(value, dict) and not marqo_index_version_lt_2_9_0:
+                for k, v in value.items():
+                    if isinstance(v, int):
+                        instance.fields.int_fields[f"{key}.{k}"] = v
+                        instance.fields.score_modifiers_fields[f"{key}.{k}"] = v
+                    elif isinstance(v, float):
+                        instance.fields.float_fields[f"{key}.{k}"] = float(v)
+                        instance.fields.score_modifiers_fields[f"{key}.{k}"] = v
             else:
                 raise VespaDocumentParsingError(f"Document {document} with field {key} has an "
```

Comment:
When a user tries to index maps with a <2.9 index, we need a better error message that says your index version doesn't support this etc

Reply:
Ok, will do so

Reply:
Done